ironfish/src/rpc/adapters/socketAdapter/socketAdapter.ts

```diff
@@ -316,4 +330,16 @@ export abstract class RpcSocketAdapter implements IRpcAdapter {
       data: data,
     }
   }
+
+  constructUnauthenticatedRequest(request: SocketRpcRequest): ServerSocketRpc {
+    const error = new Error(`Missing or bad authentication`)
+
+    const data: SocketRpcError = {
+      code: ERROR_CODES.UNAUTHENTICATION,
+      message: error.message,
+      stack: error.stack,
+    }
+
+    return this.constructMessage(request.mid, 401, data)
+  }
```

Comment:
This is for another PR, but I think if we just made a new method to construct a response to a message from an error, we could probably reuse it for this case, and the `ResponseError` on line 231.